packages/mobile/src/transactions/saga.ts

```diff
@@ -1,25 +1,130 @@
+  const localCurrencyCode = yield select(getLocalCurrencyCode)
+  const token = Token.CUsd
+
+  // Using a dynamic require import here because of an import cycle around TransactionsList that is
+  // causing many tests to fail. Alternative is to mock this entire saga file or significantly restructure things.
+  const query = require('src/transactions/TransactionsList').TRANSACTIONS_QUERY
+
+  const userTxQueryChannel: EventChannel<UserTxQueryChannelEvent> = eventChannel((emitter) => {
+    const querySubscription = apolloClient
+      .watchQuery<UserTransactionsQuery | undefined>({
+        query,
+        variables: {
+          address,
+          token,
+          localCurrencyCode,
+        },
+        fetchPolicy: 'cache-and-network',
+      })
+      .subscribe({
+        next: (queryResult) => {
+          if (queryResult.loading) {
+            return
+          }
+
+          if (queryResult.errors) {
+            Logger.error(TAG, 'Error watching user tx query' + JSON.stringify(queryResult.errors))
+            return
+          }
+          emitter({
+            transactionFeedFragments: getTxsFromUserTxQuery(queryResult.data),
+          })
+        },
+        error: (e) => console.error(e),
+      })
+
+    return querySubscription.unsubscribe
+  })
+  yield spawn(watchUserTxQueryChannel, userTxQueryChannel)
+}
+
+function* watchUserTxQueryChannel(channel: EventChannel<UserTxQueryChannelEvent>) {
+  try {
+    Logger.debug(`${TAG}/watchUserTransactionQueryChannel`, 'Started channel watching')
+    while (true) {
+      const event: UserTxQueryChannelEvent = yield take(channel)
+      if (!event) {
+        Logger.debug(`${TAG}/watchUserTransactionQueryChannel`, 'Data in channel was empty')
+        continue
+      }
+      Logger.info(`${TAG}/watchUserTransactionQueryChannel`, 'Notification received in the channel')
+      const knownTxs: KnownFeedTransactionsType = yield select(knownFeedTransactionsSelector)
+      const newTxs = event.transactionFeedFragments.filter((tx) => !knownTxs[tx.hash])
+      if (!newTxs.length) {
+        continue
+      }
+      yield put(newTransactionsInFeed(newTxs))
+      yield call(cleanupStandbyTransacitons, newTxs)
+    }
+  } catch (error) {
+    Logger.error(
+      `${TAG}/watchUserTransactionQueryChannel`,
+      'Error proccesing user tx query channel event',
+      error
+    )
+  } finally {
+    Logger.debug(`${TAG}/watchUserTransactionQueryChannel`, 'User tx channel terminated')
+  }
+}
+
+// Remove standby txs from redux state when the real ones show up in the feed
+function* cleanupStandbyTransacitons(newFeedTransactions: TransactionFeedFragment[]) {
+  const standbyTxs: StandbyTransaction[] = yield select(standbyTransactionsSelector)
+  const newFeedTxHashes = new Set(newFeedTransactions.map((tx) => tx?.hash))
+  for (const standbyTx of standbyTxs) {
+    if (
+      standbyTx.hash &&
+      standbyTx.status !== TransactionStatus.Failed &&
```

Comment:
Is there value in keeping failed transactions? If timestamp is easily accessible, maybe we can safely cleanup any failed transactions of a certain age or older?

Reply:
Hmm, good question. In theory, this case never happens b.c. if the tx failed, it'll never show up in blockscout and therefore the feed. But I'm not 100% sure so I'd rather keep the logic we had for this (previously in TransactionList)